config/rbac/role.yaml

```diff
@@ -12,6 +12,18 @@ rules:
   verbs:
   - get
   - list
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
```

Comment:
This was needed since i am testing the binding to occur on primaza main cluster. The srevice binding has to list, update and watch applications(deployments). I believe we can let go of this role if we test all scenarios on the worker cluster and then i can remove the controller reconciler added for service binding in primaza main. Makes sense?

Reply:
Have remove the role permission added for deployment, as we would be deploying the agent app controller to the requested namespace, whose service account already has the permissions to list,update deployments.